libsql/src/hrana/stream.rs

```diff
@@ -274,10 +274,21 @@ where
         Ok(responses)
     }
 
-    async fn finalize(&mut self, req: StreamRequest) -> Result<StreamResponse> {
-        let [resp, _] = self.send_requests([req, StreamRequest::Close]).await?;
+    async fn finalize(&mut self, req: StreamRequest) -> Result<(StreamResponse, bool)> {
+        let [resp, _, get_autocommit] = self
+            .send_requests([req, StreamRequest::GetAutocommit, StreamRequest::Close])
```

Comment:
aren't the results in the wrong order? 🤔 
```suggestion
        let [resp, get_autocommit, _] = self
            .send_requests([req, StreamRequest::GetAutocommit, StreamRequest::Close])
```

Reply:
mind adding a sim test?

Reply:
This is tested, hence I flipped the order. I found this pretty weird myself.